### PR TITLE
Initial Support : Automatic Transaction Rebroadcasting

### PR DESCRIPTION
**Pull request by trevelyan**
Includes some initial support for automatic transaction rebroadcasting.

We are not adding the ATR transactions to the block or validating them. But we are calculating them based on the assumption of a 2 block EPOCH. So this is scaffold code. What remains is basically to decide how to store the ATR transactions in the block so that we can have a fast and optimized sweep through the transaction set.

This PR also includes some updates to the spammer code that gets the transactions it is producing using Saito tokens owned by the wallet. And adds routing hops to them on receipt in the mempool so that we can calculate / test the routing work-related functions as needed.

I think this is ready for merge. The critical parts are largely contained to the Block.rs function that generates consensus data. And it fixes a number of bugs introduced by our merging of previous pull requests from different branches yesterday.

### CODE REVIEW

File: src/bin/spammer.rs
Comment (BearGuy) [inline src/bin/spammer.rs:50]: Overall, PR is looking good. One thing I'll request before merging this is to fix the formatting. You can see in files changed how poor the formatting can get in a bunch of places. I'm sure it looks fine in your dev environment, but for everyone else, the code becomes messy and strange. I'd like to request that you fix the spacing in your environment so that we can have proper spacing in the repo.

**Review by BearGuy — CHANGES_REQUESTED** [review]

File: src/block.rs
Comment (trevelyan) [reply src/block.rs:564]: The first random number is the miner_random.

Logic here is going to get more complicated as we add in staking, so lets come back and fix variable names when we add that!

File: src/bin/spammer.rs
Comment (trevelyan) [reply src/bin/spammer.rs:50]: I've noticed these issues. The problems were especially severe in the mempool.run() function and the GenerateTransaction logic that has now been moved into that spammer class. I suspect the "cargo fmt" formatter was struggling with the tokio/async loop internals, as well as cases within a function where we have code that looks like this -- with read / write access compartmentalized so that the lock releases quickly:

   let publickey;

  {
      let wallet = wallet_lock.write().await;
      publickey = wallet.get_publickey();
 }

The problems may also be getting caused by the Github web-interface for sorting out merges, since when the files load in that web interface they are often mangled in terms of how they display and how the different sections align. I've been using that to handle those fixes and can switch to another approach if there is something better.

At this point I've cleaned up all of those files several times. It's a bit of a disinhibitor to run "cargo fmt" when you know you might have to re-indent a lengthy function. I'm happy for us to run cargo fmt by default on code. What I try to do is clean up code when I'm looking at it in VIM and it is badly aligned. If the issue is others having a different standard for how many spaces are in a tab (I believe VIM is 8) I would be happy to switch over.

Otherwise, the best way to avoid the problem would seem to be not having large changes that sweep over multiple classes and require manual merge.

**Pull request merged**
